src/txn/txn_ckpt.c

```diff
@@ -897,7 +897,13 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
      */
     if (F_ISSET(hs_dhandle, WT_DHANDLE_OPEN)) {
         time_start_hs = __wt_clock(session);
+        conn->txn_global.checkpoint_running_hs = true;
```

Comment:
Do you need a memory barrier here?

Reply:
Interesting question - by the looks of it we can guarantee that `conn->txn_global.checkpoint_running_hs` will be set to true before we call `__wt_checkpoint()`. However, [we are not using memory barrier](https://github.com/wiredtiger/wiredtiger/blob/ba88c91101ff0ee6e24373e701b1f9537cc46f3b/src/txn/txn_ckpt.c#L1101) in `__txn_checkpoint_wrapper()` as well where we are setting/clearing `txn_global->checkpoint_running` flag.

Reply:
I do not think we need one. There are no precise timing considerations for functional correctness here. It is not that we are waiting for the exact spot in the code where we disable eviction on this file. Worse we would let a non-HS page be evicted and add some more to the HS content in the cache that can't evict.

Reply:
That's true. Good point.